core/src/lib.rs

```diff
@@ -102,25 +102,27 @@ where
 }
 
 /// Create a worker for replaying a specific history. It will auto-shutdown as soon as the history
-/// has finished being replayed. The provided client should be a mock, and this should only be used
-/// for workflow testing purposes.
-pub fn init_replay_worker(
+/// has finished being replayed.
+pub fn init_replay_worker<I>(
     mut config: WorkerConfig,
-    history: &History,
-) -> Result<Worker, anyhow::Error> {
+    histories: I,
```

Comment:
I think a cleaner API would be to return a struct here that has a worker and a public function to be given a history. That way lang can send individual histories at any time after init instead of maintaining an iterator. The "replayer" is modeled as a class in Python which makes a lot of sense and probably deserves to be a struct in Rust.